lib/lua/index.js

```diff
@@ -49,12 +51,14 @@ function loadScriptIfMissing(client, scriptKey) {
 }
 
 function buildCache(client) {
+  if (scriptsLoaded) return scriptsLoadedPromise;
+  scriptsLoaded = true;
   // We could theoretically pipeline this, but it's pretty insignificant.
-  return readScripts().then(() =>
+  return (scriptsLoadedPromise = readScripts().then(() =>
     Promise.all(
       Object.keys(shas).map((key) => loadScriptIfMissing(client, key))
     )
-  );
+  ));
```

Comment:
@skeggse is this a reasonable way to accomplish this, in lieu of a WeakMap? I see this being done above when reading the scripts in. perhaps this isn't reasonable because it's not done at the `Client` level?

Reply:
Yeah, we'll need to do it per-Queue or per-Client - otherwise we won't correctly load scripts into auxiliary clients.